dialogue-core/src/main/java/com/palantir/dialogue/core/TimingEndpointChannel.java

```diff
@@ -52,18 +54,18 @@ final class TimingEndpointChannel implements EndpointChannel {
         this.delegate = delegate;
         this.ticker = ticker;
         ClientMetrics metrics = ClientMetrics.of(taggedMetrics);
-        this.successTimer = metrics.response()
+        this.successTimer = Suppliers.memoize(() -> metrics.response()
                 .channelName(channelName)
                 .serviceName(endpoint.serviceName())
                 .endpoint(endpoint.endpointName())
                 .status("success")
-                .build();
-        this.failureTimer = metrics.response()
+                .build());
+        this.failureTimer = Suppliers.memoize(() -> metrics.response()
                 .channelName(channelName)
                 .serviceName(endpoint.serviceName())
                 .endpoint(endpoint.endpointName())
                 .status("failure")
-                .build();
+                .build());
```

Comment:
I wonder if we should provide a lazy TaggedMetricRegistry implementation which registers non-gauge metrics the first time they're updated? Probably not, since the metric implementations aren't interfaces but rather concrete implementations.

Reply:
I've debated adding a cache in generated metric-schema `*Metrics` instances to reduce the frequency of `MetricName` creation, though most places can store a reference to the metric itself, but it would help these sticky channels.